File: src/AnyOfRule.java
import java.util.List;
public class AnyOfRule implements RequirementRule {
    private List<RequirementRule> rules;

    public AnyOfRule(List<RequirementRule> rules) {
        this.rules = rules;
    }

    @Override
    public boolean isSatisfied(Student student) {
        for (RequirementRule rule : rules) {
            if (rule.isSatisfied(student)) {
                return true;
            }
        }
        return false;
    }
    
}

File: src/any_of_rule.cpp
#include "any_of_rule.hpp"

#include "student.hpp"

#include <utility>

namespace course_planner {

AnyOfRule::AnyOfRule(std::vector<RequirementRulePtr> rules)
    : rules_(std::move(rules)) {}

bool AnyOfRule::is_satisfied(const Student& student) const {
    for (const auto& rule : rules_) {
        if (rule->is_satisfied(student)) {
            return true;
        }
    }
    return false;
}

}  // namespace course_planner
